packages/node-core/src/indexer/storeModelProvider/storeModel.service.ts

```diff
@@ -0,0 +1,87 @@
+// Copyright 2020-2024 SubQuery Pte Ltd authors & contributors
+// SPDX-License-Identifier: GPL-3.0
+
+import {Injectable} from '@nestjs/common';
+import {Sequelize, Transaction} from '@subql/x-sequelize';
+import {NodeConfig} from '../../configure';
+import {getLogger} from '../../logger';
+import {exitWithError} from '../../process';
+import {BaseStoreModelService} from './baseStoreModel.service';
+import {CsvStoreService} from './csvStore.service';
+import {MetadataModel} from './metadata/metadata';
+import {METADATA_ENTITY_NAME} from './metadata/utils';
+import {IModel} from './model';
+import {PlainModel} from './model/model';
+import {PlainPoiModel, POI_ENTITY_NAME} from './poi';
+import {IStoreModelProvider} from './types';
+
+const logger = getLogger('PlainStoreModelService');
+
+@Injectable()
+export class PlainStoreModelService extends BaseStoreModelService implements IStoreModelProvider {
+  constructor(
+    private sequelize: Sequelize,
+    private config: NodeConfig
+  ) {
+    super();
+  }
+
+  get metadata(): MetadataModel {
+    if (!this.cachedModels[METADATA_ENTITY_NAME]) {
+      if (!this.metadataRepo) {
+        throw new Error('Metadata entity has not been set');
+      }
+      this.cachedModels[METADATA_ENTITY_NAME] = new MetadataModel(this.metadataRepo) as any;
+    }
+
+    return this.cachedModels[METADATA_ENTITY_NAME] as unknown as MetadataModel;
+  }
+
+  get poi(): PlainPoiModel | null {
+    if (!this.cachedModels[POI_ENTITY_NAME]) {
+      if (!this.poiRepo) {
+        return null;
+        // throw new Error('Poi entity has not been set on store cache');
+      }
+      this.cachedModels[POI_ENTITY_NAME] = new PlainPoiModel(this.poiRepo) as any;
+    }
+
+    return this.cachedModels[POI_ENTITY_NAME] as unknown as PlainPoiModel;
+  }
+
```

Comment:
Could this be moved to the base service?

Reply:
BaseCacheService currently extends BaseService, and I think it should not be moved under baseService.